sql/delete.go

```diff
@@ -18,11 +18,56 @@
 package sql
 
 import (
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/sql/parser"
-	"github.com/cockroachdb/cockroach/util"
 )
 
-// Delete TODO(pmattis): document.
+// Delete deletes rows from a table.
 func (p *planner) Delete(n *parser.Delete) (planNode, error) {
-	return nil, util.Errorf("TODO(pmattis): unimplemented")
+	tableDesc, err := p.getAliasedTableDesc(n.Table)
+	if err != nil {
+		return nil, err
+	}
+
+	primaryKeyPrefix := encodeIndexKeyPrefix(tableDesc.ID, tableDesc.Indexes[0].ID)
+
+	b := client.Batch{}
+
+	// TODO(tamird,pmattis): avoid going through Select to avoid encoding
+	// and decoding keys. Also, avoiding Select may provide more
+	// convenient access to index keys which we are not currently
+	// deleting.
+	node, err := p.Select(&parser.Select{Exprs: parser.SelectExprs{
+		&parser.StarExpr{TableName: tableDesc.Name},
+	}, From: parser.TableExprs{n.Table}, Where: n.Where})
```

Comment:
Can you reformat this? The current formatting makes it hard to see the structure.

```
  node.err := p.Select(&parser.Select{
    Exprs: parser.SelectExprs {
      &parser.StarExpr{TableName: tableDesc.Name},
    },
    From: parser.TableExprs{n.Table},
    Where: n.Where,
  })
```